k9mail/src/main/java/com/fsck/k9/mailstore/LocalStore.java

```diff
@@ -789,6 +789,13 @@ private static Part findPartById(Part searchRoot, long partId) {
 
             if (part instanceof LocalMessage) {
```

Comment:
Since this can only be true for the root message you might as well move it outside the loop.